src/TSTransformer/util/spreadDestruction/spreadDestructObject.ts

```diff
@@ -0,0 +1,53 @@
+import luau from "@roblox-ts/luau-ast";
+import { TransformState } from "TSTransformer/classes/TransformState";
+
+export function spreadDestructObject(
+	state: TransformState,
+	parentId: luau.AnyIdentifier,
+	preSpreadNames: Array<luau.Expression>,
+) {
+	const extracted = state.pushToVar(
+		luau.set(
+			preSpreadNames.map(expression => {
+				if (luau.isPropertyAccessExpression(expression)) return luau.string(expression.name);
+				if (luau.isComputedIndexExpression(expression)) return expression.index;
```

Comment:
`luau.isComputedIndexExpression(expression)` case appears to be not covered by unit tests